client/src/components/Nav.js

```diff
@@ -1,22 +1,63 @@
 import styled from "styled-components";
+import React, { useRef, useState } from "react";
+import { useOnClickOutside } from "../hooks";
 
+import SideNav from "./SideNav";
+import Profile from "./Profile";
+import Group from "./Group";
 import Logo from "./Logo";
 import IconUser from "./IconUser";
 import IconGroup from "./IconGroup";
 
 export default function Nav() {
+  const [openNav, setOpenNav] = useState(false);
+  const [openProfile, setOpenProfile] = useState(false);
+  const [openGroup, setOpenGroup] = useState(false);
+
+  const nodeNav = useRef();
+  useOnClickOutside(nodeNav, () => setOpenNav(false));
+
+  const nodeProfile = useRef();
+  useOnClickOutside(nodeProfile, () => setOpenProfile(false));
+
+  const nodeGroup = useRef();
+  useOnClickOutside(nodeGroup, () => setOpenGroup(false));
+
   return (
-    <Navi>
-      <Logo />
-      <GroupBox>
-        <IconGroup />
-        <NavLink>groupname</NavLink>
-      </GroupBox>
-      <GroupBox>
-        <IconUser />
-        <NavLink>username</NavLink>
-      </GroupBox>
-    </Navi>
+    <>
```

Comment:
No need for the additional React Fragments as the `<Navi>` wraps around the whole content